docs/maintainers/niche-package-maintenance/rustc/rust-packaging-faq.md

```diff
@@ -0,0 +1,263 @@
+
+Debian maintainers do make exceptions for extremely valuable software, such as Firefox.
+The general approach for these exceptions is to use special settings to temporarily configure the compiler to allow unstable features in a stable version.
+Ubuntu packages Firefox as a Snap, which avoids the need for special hacks like this.
+
+(debian-rust-tools)=
+## What Rust-specific tooling do maintainers use, and how does it fit into the wider ecosystem of packaging tools?
+
+The ecosystem of Debian packaging tools is big and diverse, with a great deal of overlap and intended deprecation.
+This section focuses only on those tools that are likely to be useful when working on Rust code.
+A much more comprehensive reference is available in [the Debian developers-reference](https://www.debian.org/doc/manuals/developers-reference/tools.en.html).
+
+There are, roughly speaking, three phases to creating a `.deb` package.
+
+1. Create the source package.
+2. Build the binary package from the source package.
+3. Test and upload the binary package.
+
+Although coarse, keeping this in mind can help build a more coherent mental model of how the tools interoperate.
+
+### Tools for creating and modifying the source package
+
+**Language-agnostic tools:**
+
+`uscan`
+: Short for "upstream scan", it automates tasks related to keeping up-to-date with upstream code.
+It relies on you creating a `debian/watch` file, and then can look for new releases, grab them, and verify signatures.
+Source packages require "orig" tarballs as a component, which are the unmodified upstream source code, and `uscan` can help you grab them.
+
+`quilt`
+: An older, but still widely-used tool to manage a sequence of patches that apply to the upstream source.
+It can not only help you manage the order of patches and their current state of application, but can help you generate patches from modified source code.
+Some maintainers prefer to manage patches with git-based tooling, which didn't exist when `quilt` was first developed.
+In theory, a `git` branch could encode the order of the patches, which are then rebased onto the new code when doing an update.
+Nevertheless, being familiar with `quilt` is helpful when understanding packages worked on by multiple maintainers.
+
+`dch`
+: Automation for managing the `/debian/changelog` file.
+It helps you generate timestamps, author information, and more.
+
+`git-ubuntu`
+: An extension to `git` that can clone the repository attached to a package directly from Launchpad.
+It has many other features intended to enable more complex package workflows, which are described in the [Ubuntu Maintainer's Handbook](https://github.com/canonical/ubuntu-maintainers-handbook).
+
+`pull-lp-source`
+: Similar to `git-ubuntu`, which should be used instead where possible.
+The main difference is that instead of cloning the `git` repository, it grabs the source package from Launchpad.
+That means you'll get things like the orig tarball, but not the `git` history.
```

Comment:
While useful and a nice overview, this seems out of scope for this FAQ. Could we perhaps incorporate it into https://documentation.ubuntu.com/project/contributors/setup/ (either as a separate article, or directly into https://documentation.ubuntu.com/project/contributors/setup/set-up-for-ubuntu-development/)?